WalletWasabi/BitcoinCore/CachedRpcClient.cs

```diff
@@ -0,0 +1,208 @@
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Primitives;
+using NBitcoin;
+using NBitcoin.RPC;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace WalletWasabi.BitcoinCore
+{
+	public class CachedRpcClient : RpcClientBase
+	{
+		public CachedRpcClient(RPCClient rpc, IMemoryCache cache)
+			: base(rpc)
+		{
+			Cache = cache;
+		}
+		
+		public IMemoryCache Cache { get; }
+
+		public override async Task<uint256> GetBestBlockHashAsync()
+		{
+			uint256 blockHash;
```

Comment:
Variable declaration can be inlined. Remove this line and add the declaration like so:
`if (!Cache.TryGetValue(cacheKey, out uint256 blockHash))`

Same for similar variables in other methods.